pkg/serviceaccount/tokens_controller.go

```diff
@@ -336,22 +337,31 @@ func (e *TokensController) generateTokenIfNeeded(serviceAccount *api.ServiceAcco
 		secret.Data = map[string][]byte{}
 	}
 
-	tokenData, ok := secret.Data[api.ServiceAccountTokenKey]
-	if ok && len(tokenData) > 0 {
+	caData := secret.Data[api.ServiceAccountRootCAKey]
+	needsCA := len(e.rootCA) > 0 && bytes.Compare(caData, e.rootCA) != 0
+
+	tokenData := secret.Data[api.ServiceAccountTokenKey]
+	needsToken := len(tokenData) == 0
```

Comment:
it would be ideal, though it would mean adding token validation into the token controller. I like the idea, but wanted to think through it more, and separately from the CA populate part.